homeassistant/components/reolink/host.py

```diff
@@ -359,6 +367,43 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _async_poll_all_motion(self, *_) -> None:
+        """Poll motion and AI states until the first ONVIF push is received."""
+        if self._webhook_reachable.is_set():
+            # ONVIF push is working, stop polling
+            self._cancel_poll = None
+            return
+
+        try:
+            await self._api.get_motion_state_all_ch()
+        except (
+            aiohttp.ClientConnectorError,
+            ReolinkError,
+        ) as err:
+            _LOGGER.error(
+                "Reolink error while polling motion state for host %s:%s: %s",
+                self._api.host,
+                self._api.port,
+                str(err),
+            )
+        except asyncio.TimeoutError:
+            _LOGGER.error(
+                "Reolink timeout error while polling motion state for host %s:%s",
+                self._api.host,
+                self._api.port,
+            )
+
+        # After receiving the new motion states in the upstream lib,
+        # update the binary sensors with async_write_ha_state
+        # The same dispatch as for the webhook can be used
+        async_dispatcher_send(self._hass, f"{self.webhook_id}_all", {})
+
+        # schedule next poll
+        if not self._hass.is_stopping:
+            self._cancel_poll = async_call_later(
+                self._hass, POLL_INTERVAL_NO_PUSH, self._poll_job
+            )
```

Comment:
Should this happen in a finally block in case `await self._api.get_motion_state_all_ch()` throws an exception that is unexpected so we keep trying?

Reply:
Good idea, done